pymc3/gp/gp.py

```diff
@@ -68,17 +125,231 @@ def _random(mean, cov, size=None):
+
+        L = cholesky(stabilize(Kxx))
+        A = solve_lower(L, Kxz)
+
+        cov = Kzz - tt.dot(tt.transpose(A), A)
+        mean = self.m(Z) + tt.squeeze(tt.transpose(tt.dot(tt.transpose(A), self.v)))
+        #mean = self.m(Z) + tt.dot(tt.transpose(A), self.v)
+        return mean, stabilize(cov)
+
+    @property
+    def RV(self):
+        self.L = cholesky(stabilize(self.K(self.X)))
+        # i get an interesting error when i use the name _rotated__ with two underscores
+        self.v = Normal(self.name + "_rotated_", mu=0.0, sd=1.0, shape=self.nf)
+        self.f = Deterministic(self.name, tt.dot(self.L, self.v))
+        return self.f
+
+
+class GPFullConjugate(GPBase, Continuous):
+    """Gausian process
+
+    Parameters
+    ----------
+    X : array
+        Array of predictor variables.
+    mean_func : Mean
+        Mean function of Gaussian process
+    cov_func : Covariance
+        Covariance function of Gaussian process
+    cov_func_noise : Covariance
+        Covariance function of noise Gaussian process
+    """
+    def __init__(self, X, mean_func, cov_func, cov_func_noise, *args, **kwargs):
+        self.X = X
+        self.nf = self.X.shape[0]
+        self.K = cov_func
+        self.Kn = cov_func_noise
+        self.m = mean_func
+        self.mean = self.mode = self.m(X)
+
+        kwargs.setdefault("shape", X.squeeze().shape)
+        super(GPFullConjugate, self).__init__(*args, **kwargs)
+
+    def prior(self, obs_noise=False):
+        mean = self.m(self.X)
+        if obs_noise:
+            cov = self.K(self.X) + self.Kn(self.X)
+        else:
+            cov = stabilize(self.K(self.X))
+        return mean, cov
+
+    def conditional(self, Z, y, obs_noise=False):
+        nz = Z.shape[0]
+        Kxx = self.K(self.X)
+        Knx = self.Kn(self.X)
+        Kxz = self.K(self.X, Z)
+        Kzz = self.K(Z)
+
+        r = y - self.m(self.X)
+        L = cholesky(stabilize(Kxx) + Knx)
+        A = solve_lower(L, Kxz)
+        V = solve_lower(L, r)
+        mean = tt.dot(tt.transpose(A), V) + self.m(Z)
+        if obs_noise:
+            cov = self.Kn(Z) + Kzz - tt.dot(tt.transpose(A), A)
+        else:
+            cov = stabilize(Kzz - tt.dot(tt.transpose(A), A))
+        return mean, cov
+
+    def logp(self, y):
+        mean = self.m(self.X)
+        L = cholesky(stabilize(self.K(self.X)) + self.Kn(self.X))
+        return MvNormal.dist(mu=mean, chol=L).logp(y)
 
-        return MvNormal.dist(mu, Sigma).logp(Y)
 
+class GPSparseConjugate(GPBase, Continuous):
+    """Sparse Gausian process approximation
 
-def sample_gp(trace, gp, X_values, samples=None, obs_noise=True, model=None, random_seed=None, progressbar=True,
-              chol_const=True):
+    Parameters
+    ----------
+    X : array
+        Array of predictor variables.
+    mean_func : Mean
+        Mean function of Gaussian process
+    cov_func : Covariance
+        Covariance function of Gaussian process
+    sigma2 : scalar or array
```

Comment:
i hadnt really thought about it.  thinking about it more, yeah I think your right it makes more sense to parameterize in terms of sigma, not sigma2